src/RL/nn_structures/heterogeneousfullfeaturedcpnn.jl

```diff
@@ -21,36 +21,27 @@ Flux.@functor HeterogeneousFullFeaturedCPNN
 
 # TODO make it possible to use global features
 function (nn::HeterogeneousFullFeaturedCPNN)(states::BatchedHeterogeneousTrajectoryState)
-    @assert !isnothing(states.allValuesIdx)
-
     variableIdx = states.variableIdx
     batchSize = length(variableIdx)
-    allValuesIdx = states.allValuesIdx
-    actionSpaceSize = size(allValuesIdx, 1)
+    actionSpaceSize = size(states.fg.valnf, 2)
     mask = device(states) == Val(:gpu) ? CUDA.zeros(Float32, 1, actionSpaceSize, batchSize) : zeros(Float32, 1, actionSpaceSize, batchSize) # this mask will replace `reapeat` using broadcasted `+`
 
     # chain working on the graph(s) with the GNNs
     featuredGraph = nn.graphChain(states.fg)
-    nodeFeatures = featuredGraph.nf # FxNxB
-    variableFeature = fg.varnf[:, variableIdx]
-    valueFeatures = fg.valnf[:, possibleValuesIdx]
+    variableFeatures = featuredGraph.varnf # FxNxB
+    valueFeatures = featuredGraph.valnf
     globalFeatures = featuredGraph.gf # GxB
 
     # Extract the features corresponding to the varibales
     variableIndices = nothing
     Zygote.ignore() do
         variableIndices = Flux.unsqueeze(CartesianIndex.(variableIdx, 1:batchSize), 1)
     end
-    variableFeatures = nodeFeatures[:, variableIndices] # Fx1xB
-    variableFeatures = reshape(nn.nodeChain(RL.flatten_batch(variableFeatures)), :, 1, batchSize) # F'x1xB
+    branchingVariableFeatures = variableFeatures[:, variableIndices] # Fx1xB
+    relevantVariableFeatures = reshape(nn.nodeChain(RL.flatten_batch(branchingVariableFeatures)), :, 1, batchSize) # F'x1xB
```

Comment:
We could potentially try it but I think we should limit the number of weights in our neural networks:  `HeterogeneousGraphConv` already deals differently with variables and values, so it might be unnecessary to further maintain differences in treatment.

Reply:
Well, the pipeline must be heterogeneous at every step, otherwise, all the work done to increase representation on the convolution will be bottlenecked by this inconsistent layer.  Let's discuss that with the others !

@gostreap @malikattalah @ziadelassal

Reply:
I have another comment, what if the dimensions of the variable embedding and the value embedding do not match? This requires that the GNN satisfy this condition.